ignite/pkg/protoanalysis/protoutil/creator.go

```diff
@@ -0,0 +1,636 @@
+		enum.Elements = append(enum.Elements, opt)
+	}
+	for _, field := range e.fields {
+		enum.Elements = append(enum.Elements, field)
+	}
+	return enum
+}
+
+// OneOfField holds information relevant to the oneof field statement.
+type OneOfFieldSpec struct {
+	name, typ string
+	sequence  int
+	options   []*proto.Option
+}
+
+// OneOfFieldOpts is a type alias for a callable accepting a OneOfField.
+type OneOfFieldOpts func(f *OneOfFieldSpec)
+
+// WithOneOfFieldOptions adds options to the oneof field.
+func WithOneOfFieldOptions(options ...*proto.Option) OneOfFieldOpts {
+	return func(f *OneOfFieldSpec) {
+		f.options = append(f.options, options...)
+	}
+}
+
+// NewOneOfField creates a new oneof field statement node:
+//
+//		// Needs to placed in a oneof block.
+//	 // int32 Foo = 1;
+//	 field := NewOneOfField("Foo", "int32", 1)
+//
+// Additional options can be created and attached to the field to the field via
+// WithOneOfFieldOptions:
+//
+//	// int32 Foo = 1 [option (foo) = 1];
+//	field := NewOneOfField("Foo", "int32", 1, WithOneOfFieldOptions(NewOption("foo", "1")))
+func NewOneOfField(typ, name string, sequence int, opts ...OneOfFieldOpts) *proto.OneOfField {
+	f := OneOfFieldSpec{name: name, typ: typ, sequence: sequence}
+	for _, opt := range opts {
+		opt(&f)
+	}
+	field := &proto.OneOfField{
+		Field: &proto.Field{
+			Name:     f.name,
+			Sequence: f.sequence,
+			Type:     f.typ,
+			Options:  []*proto.Option{},
+		},
+	}
+	field.Options = append(field.Options, f.options...)
+	return field
+}
+
+// OneOfSpec holds information relevant to the enum statement.
+type OneofSpec struct {
+	name    string
+	options []*proto.Option
+	fields  []*proto.OneOfField
+}
+
+// OneOfSpecOpts is a type alias for a callable accepting a OneOfSpec.
+type OneOfSpecOpts func(o *OneofSpec)
+
+// WithOneOfOptions adds options to the oneof.
+func WithOneOfOptions(options ...*proto.Option) OneOfSpecOpts {
+	return func(o *OneofSpec) {
+		o.options = append(o.options, options...)
+	}
+}
+
+// WithOneOfFields adds fields to the oneof.
+func WithOneOfFields(fields ...*proto.OneOfField) OneOfSpecOpts {
+	return func(o *OneofSpec) {
+		o.fields = append(o.fields, fields...)
+	}
+}
+
+// NewOneOf creates a new oneof statement node:
+//
+//	// oneof Foo {
+//	//  int32 Foo = 1;
+//	// }
+//	oneof := NewOneOf("Foo", WithOneOfFields(NewOneOfField("Foo", "int32", 1)))
+//
+// No options are attached by default, use WithOneOfOptions to add them as required.
+func NewOneOf(name string, opts ...OneOfSpecOpts) *proto.Oneof {
+	o := OneofSpec{name: name}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	oneof := &proto.Oneof{
+		Name: o.name,
+	}
+	for _, opt := range o.options {
+		oneof.Elements = append(oneof.Elements, opt)
+	}
+	for _, field := range o.fields {
+		oneof.Elements = append(oneof.Elements, field)
+	}
+	return oneof
+}
+
+// Handle this better. Currently s with at least one digit is
+// considered a number and if not, a string.
```

Comment:
```suggestion
// TODO: Handle this better. Currently s with at least one digit is
// considered a number and if not, a string.
```

Have you consider using `strconv.ParseBool` and/or `strconv.ParseInt` ?

Reply:
will go with the ParseInt here and remove the TODO.